From: Andrea Bolognani <eof@kiyuko.org>
Date: Mon, 2 Nov 2009 00:06:06 +0100
Subject: Adapt libswmhack

libswmhack tries to dlopen() several libraries (libxcb.so,
libX11.so and libXt.so), but on Debian those are symlinks
shipped in the corresponding -dev package.

Use the versioned library names (libxcb.so.1, libX11.so.6
and libXt.so.6) instead. This ensures that libswmhack will
work with just the runtime packages installed.

Bug-Debian: https://bugs.debian.org/551732
Bug: https://github.com/conformal/spectrwm/pull/63
---
 lib/swm_hack.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/swm_hack.c b/lib/swm_hack.c
index cbd2141..53c0401 100644
--- a/lib/swm_hack.c
+++ b/lib/swm_hack.c
@@ -81,7 +81,7 @@ xcbsym(const char *symbol, char **e)
 #endif
 	if (p == NULL) {
 		if (lib_xcb == NULL)
-			lib_xcb = dlopen("libxcb.so", RTLD_GLOBAL | RTLD_LAZY);
+			lib_xcb = dlopen("libxcb.so.1", RTLD_GLOBAL | RTLD_LAZY);
 		if (lib_xcb)
 			p = dlsym(lib_xcb, symbol);
 		if (p == NULL && e)
@@ -102,7 +102,7 @@ xlibsym(const char *symbol, char **e)
 #endif
 	if (p == NULL) {
 		if (lib_xlib == NULL)
-			lib_xlib = dlopen("libX11.so", RTLD_GLOBAL | RTLD_LAZY);
+			lib_xlib = dlopen("libX11.so.6", RTLD_GLOBAL | RTLD_LAZY);
 		if (lib_xlib)
 			p = dlsym(lib_xlib, symbol);
 		if (p == NULL && e)
@@ -123,7 +123,7 @@ xtlibsym(const char *symbol, char **e)
 #endif
 	if (p == NULL) {
 		if (lib_xtlib == NULL)
-			lib_xtlib = dlopen("libXt.so", RTLD_GLOBAL | RTLD_LAZY);
+			lib_xtlib = dlopen("libXt.so.6", RTLD_GLOBAL | RTLD_LAZY);
 		if (lib_xtlib)
 			p = dlsym(lib_xtlib, symbol);
 		if (p == NULL && e)
